src/coreclr/jit/assertionprop.cpp

```diff
@@ -2963,7 +2963,7 @@ GenTree* Compiler::optCopyAssertionProp(AssertionDsc*        curAssertion,
 
     // Extract the matching lclNum and ssaNum.
     const unsigned copyLclNum = (op1.lcl.lclNum == lclNum) ? op2.lcl.lclNum : op1.lcl.lclNum;
-    unsigned       copySsaNum = BAD_VAR_NUM;
+    unsigned       copySsaNum = SsaConfig::RESERVED_SSA_NUM;
```

Comment:
Is this change here the minimal fix?

Reply:
The `SetLclNum` call below resets the SSA number, so while this was clearly bogus, it did not cause any issues (because the order was SetSsaNum -> SetLclNum before). However, we only call this function during local assertion prop and swapping the order below did not have any diffs.

Reply:
The actual fix is one of the guards added in `optComputeLoopSideEffectsOfBlock`.